src/main/java/pl/edu/agh/iet/katabank/Account.java

```diff
@@ -42,12 +42,22 @@ public void setBalance(BigDecimal balance) {
     }
 
     public void withdraw(BigDecimal amount) throws IllegalArgumentException {
-        if (amount == null || amount.signum()<=0){
-            throw new IllegalArgumentException("Incorrect amount to withdraw: " + (amount == null ? null : amount.toString()));
-        }
-        if (amount.compareTo(this.balance)>0){
+        checkValidAmount(amount);
+        if (amount.compareTo(this.balance) > 0) {
             throw new IllegalArgumentException("The amount to withdraw is greater than account balance.");
         }
         this.balance = this.balance.subtract(amount);
     }
+
+    public void deposit(BigDecimal amount) {
+        checkValidAmount(amount);
+        this.balance = this.balance.add(amount);
+    }
+
+    private void checkValidAmount(BigDecimal amount) {
+        if (amount == null || amount.signum() <= 0) {
+            throw new IllegalArgumentException("Incorrect amount to withdraw: "
+                    + (amount == null ? null : amount.toString()));
+        }
```

Comment:
I like the idea of pushing the check to a separate method. Bus as we use it both for withdraw and deposit, the error message could state something like "Incorrect amount: " or "Incorrect amount to process: ", without the "withdraw" word.

Reply:
If we change this, we'll need to change that in the groovy tests for withdraw and deposit.